core/services/ocr2/plugins/functions/integration_tests/internal/testutils.go

```diff
@@ -385,23 +409,44 @@ func StartNewMockEA(t *testing.T) *httptest.Server {
 		require.NoError(t, err)
 		var jsonMap map[string]any
 		require.NoError(t, json.Unmarshal(b, &jsonMap))
-		data := jsonMap["data"].(map[string]any)
-		require.Equal(t, functions.LanguageJavaScript, int(data["language"].(float64)))
-		require.Equal(t, functions.LocationInline, int(data["codeLocation"].(float64)))
-		require.Equal(t, functions.LocationRemote, int(data["secretsLocation"].(float64)))
-		require.Equal(t, DefaultSecretsBase64, data["secrets"].(string))
-		args := data["args"].([]interface{})
-		require.Equal(t, 2, len(args))
-		require.Equal(t, DefaultArg1, args[0].(string))
-		require.Equal(t, DefaultArg2, args[1].(string))
-		source := data["source"].(string)
+		var responsePayload []byte
+		if jsonMap["endpoint"].(string) == "lambda" {
+			responsePayload = mockEALambdaExecutionResponse(t, jsonMap)
+		} else if jsonMap["endpoint"].(string) == "fetcher" {
+			responsePayload = mockEASecretsFetchResponse(t, jsonMap)
+		} else {
+			require.Fail(t, "unknown external adapter endpoint '%s'", jsonMap["endpoint"].(string))
+		}
 		res.WriteHeader(http.StatusOK)
-		// prepend "0xab" to source and return as result
-		_, err = res.Write([]byte(fmt.Sprintf(`{"result": "success", "statusCode": 200, "data": {"result": "0xab%s", "error": ""}}`, source)))
+		_, err = res.Write(responsePayload)
 		require.NoError(t, err)
 	}))
 }
 
+func mockEALambdaExecutionResponse(t *testing.T, request map[string]any) []byte {
+	data := request["data"].(map[string]any)
+	require.Equal(t, functions.LanguageJavaScript, int(data["language"].(float64)))
+	require.Equal(t, functions.LocationInline, int(data["codeLocation"].(float64)))
+	require.Equal(t, functions.LocationRemote, int(data["secretsLocation"].(float64)))
+	if data["secrets"] != DefaultSecretsBase64 && request["nodeProvidedSecrets"] != fmt.Sprintf(`{"0x0":"%s"}`, DefaultSecretsBase64) {
+		assert.Fail(t, "expected secrets or nodeProvidedSecrets to be '%s'", DefaultSecretsBase64)
+	}
+	args := data["args"].([]interface{})
+	require.Equal(t, 2, len(args))
+	require.Equal(t, DefaultArg1, args[0].(string))
+	require.Equal(t, DefaultArg2, args[1].(string))
+	source := data["source"].(string)
+	// prepend "0xab" to source and return as result
+	return []byte(fmt.Sprintf(`{"result": "success", "statusCode": 200, "data": {"result": "0xab%s", "error": ""}}`, source))
+}
+
+func mockEASecretsFetchResponse(t *testing.T, request map[string]any) []byte {
+	data := request["data"].(map[string]any)
+	require.Equal(t, "fetchThresholdEncryptedSecrets", data["requestType"], "expected requestType to be 'fetchThresholdEncryptedSecrets' but got '%s'", data["requestType"])
```

Comment:
You don't need to write out those messages. The test functions do it for you when they fail.